### ISSUE TITLE
Fix Accessibility Issues in UX Section of the Documentation Site

### ISSUE THREAD

**Issue by brea11y** [department-of-veterans-affairs/va-mobile-app#8450]
During the accessibility audit of the documentation site, issues were discovered within the UX area. Those issues can be found [in the Google sheet](https://docs.google.com/spreadsheets/d/1agEqXg74Cj2K4mEw719qwQdqUrQL7ipSOEPT1IiDCFI/edit?usp=sharing) and in the list below.

 - [x] **Page:** Accessibility checklist for content designers
    -  **Severity:** Moderate
    -  **Issue:** Missing form labels
   - **Recommendation:** The checkboxes are coming back as a missing form labels, but it is also meant to be a checkbox that you can interact with. It doesn't look like the checkboxes are actually working as intended, so let's swap these out for lists or talk to ENG about making the checkboxes function.
 - [x] **Page:** Accessibility checklist for UX designers
   -  **Severity:** Moderate
   -  **Issue:** Missing form labels
   -  **Recommendation:** The checkboxes are coming back as a missing form labels, but it is also meant to be a checkbox that you can interact with. It doesn't look like the checkboxes are actually working as intended, so let's swap these out for lists or talk to ENG about making the checkboxes function.
 - [x] **Page:** Native Mobile App Accessibility Resources & Articles
   -  **Severity:** Moderate
   -  **Issue:** Identify when links takes you to a PDF
   -  **Recommendation:** 4 of the links on the page take you to a PDF. Add "(pdf)" at the end of the links.
 - [x] **Page:** Native Mobile App Accessibility Resources & Articles	
   -  **Severity:** Minor
   -  **Issue:** Verify that YouTube videos have captions
   -  **Recommendation:** Since 6 of the links on the page take you to YouTube videos, verify that YouTube videos linked have captions present.
 - [x] **Page:** Creating components in Figma
   - **Severity:** Moderate
   - **Issue:** Skipped heading level
   - **Recommendation:** "How to set up components" should be an H2. It is currently an H3. Review heading levels to ensure that none are skipped.
 - [x] **Page:** Permissions and sharing in Figma
   - **Severity:** Serious
   - **Issue:** The table is not fully accessible
   - **Recommendation:** Revise the table to be more accessible to screenreader users (i.e. ensure the top row are labeled as <th>, etc.)
 - [x] **Page:** Permissions and sharing in Figma
   - **Severity:** Moderate
   - **Issue:** Redundant links
   - **Recommendation:** The role links in the table are redundant. Since "Roles" is linked, I'd consider removing the links from the individual roles themselves.

**Comment by brea11y** [comment]
Updates in PR #8457
